pygt3x/reader.py

```diff
@@ -297,10 +301,22 @@ def _get_data(self, num_rows=None):
             self.temperature = np.concatenate(temperature)
 
         # Make sure each second appears sample rate times
+        # 1) check for and remove identical samples
+        self.acceleration, counts = np.unique(self.acceleration, axis=0, return_counts=True)
+        duplicates_removed = self.acceleration[counts > 1]
+        if duplicates_removed.size > 0:
+            self.logger.warning(f"{duplicates_removed.shape[0]} duplicate accelerometer samples removed.")
+            for d in duplicates_removed:
+                self.logger.debug(f"Duplicate sample removed: {d.tolist()}")
+
```

Comment:
I am unsure about this. We can have bouts of 0 continuous values for the accelerometer if you leave the device unmoved. Plus we only have 12 bits of data for each axis so duplicates are likely. Am I misunderstanding this?

Reply:
This code checks for and removes "samples" from the acceleration array that are identical - that is, they have the exact same timestamp, x, y, and z values. 

For further explanation, as per the initial comment above, the code below this, now on lines 313-317, was checking for seconds of acceleration data that had the incorrect number of samples and displaying a warning. After investigating multiple instances where this warning occurred, I found that each time was because an entire log record (one second of accelerometer data) was being written twice resulting in an entire second of duplicate records. This code was inserted to remove these exact duplicates. The original code was kept in case there is another situation I have not observed that could cause a second to have the incorrect number of samples.

Reply:
I think there is a bug in here. But the time you get to the call to `np.unique`, `self.acceleration` is a concatenation of all records. I will push a fix to the new PR.

Reply:
I don't think that is a bug, this was done by design as it was inserted prior to the check for seconds with incorrect number of samples (which was retained in case there were other cases it would capture that I had not observed). There are a number of ways/places that this check could be performed. This seemed like the logical place for it. Checking each log record as it is read in would likely be less efficient in terms of processing time, but could prevent the record from ever being concatenated into the acceleration array.

Reply:
I will check the code in the new PR.